### PR TITLE
Transferring ethers with privacy protection based on precompiled cont…

### PR DESCRIPTION
**Pull request by JackXLu**
Transferring ethers with privacy protection based on precompiled contracts for ring signature verification.

### CODE REVIEW

File: EIPS/eip-draft_Precompiled+contracts+for+ring+signature+verification+on+the+elliptic+curve+alt_bn128__abbrev.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+## Preamble
+
+    EIP:
```
Comment (pirapira) [inline EIPS/eip-draft_Precompiled+contracts+for+ring+signature+verification+on+the+elliptic+curve+alt_bn128__abbrev.md]: Please use 701 and move this file to EIPS/eip-701.md

File: EIPS/eip-draft_Precompiled+contracts+for+ring+signature+verification+on+the+elliptic+curve+alt_bn128__abbrev.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+*E*: an elliptic curve equation;  
+*G*: a base point;  
+*Hs*: a cryptographic hash function which maps string of {0,1} to element of *F_q*;  
+*Hp*: a hash function which creates a mapping between elements of *E(F_q)*;  
++: point addition on elliptic curve;  
+*: scalar multiplication on elliptic curve.  
+
+**Ring signature verification**  
+
+Signature σ consists of a set of public keys, two arrays of random numbers selected by signer and a point I called key image.
```
Comment (HarryR) [inline EIPS/eip-draft_Precompiled+contracts+for+ring+signature+verification+on+the+elliptic+curve+alt_bn128__abbrev.md]: The Franklin, Zhang signature algorithm and its relation to coin tumbling is described in the following IACR paper: https://eprint.iacr.org/2017/881.pdf

Implementations in Solidity and Go for the Ethereum alt_bn128 curve are located at https://github.com/clearmatics/mobius and https://github.com/clearmatics/orbital

File: EIPS/eip-draft_Transferring+ethers+with+privacy+protection+using+precompiled+contracts+for+ring+signature+verification__abbrev.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+## Motivation
+Currently, ether transactions on Ethereum are fully transparent, which makes them unsuitable for several use-cases where users’ privacy is of concern and the link between sender and receiver should not be retrieved by anyone else. #196 and #197 suggest adding precompiled contracts for addition and scalar multiplication on a specific pairing-friendly elliptic curve to verify zkSNARKs in Ethereum smart contract, thus increasing the privacy for users. But it has high computation complexity. Monero solves this problem by using Cryptonote under UTXO model. However, this technology cannot be directly used in Ethereum under account model. Considering all of the above, this EIP forgoes zero knowledge proof and proposes to implement a special precompiled contract for ring signature verification and add one-time account mechanism on Ethereum to provide privacy protection for ether transactions. Note that, with this smart contract, users can choose regular transactions or transactions with privacy protection depending on their needs for privacy and anonymity.
```
Comment (HarryR) [inline EIPS/eip-draft_Transferring+ethers+with+privacy+protection+using+precompiled+contracts+for+ring+signature+verification__abbrev.md]: Account abstraction discussed in #678 hopes to provide a mechanism to address these problems, I hope that this precompiled contract will be a useful building block to solve problems like this.

**Comment by Arachnid** [comment]
This is a courtesy notice to let you know that the format for EIPs has been modified slightly. If you want your draft merged, you will need to make some small changes to how your EIP is formatted:

 - Frontmatter is now contained between lines with only a triple dash ('---')
 - Headers in the frontmatter are now lowercase.

If your PR is editing an existing EIP rather than creating a new one, this has already been done for you, and you need only rebase your PR.

In addition, a continuous build has been setup, which will check your PR against the rules for EIP formatting automatically once you update your PR. This build ensures all required headers are present, as well as performing a number of other checks.

Please rebase your PR against the latest master, and edit your PR to use the above format for frontmatter. For convenience, here's a sample header you can copy and adapt:

```
---
eip: <num>
title: <title>
author: <author>
type: [Standards Track|Informational|Meta]
category: [Core|Networking|Interface|ERC] (for type: Standards Track only)
status: Draft
created: <date>
---
```

File: EIPS/eip-draft_Transferring+ethers+with+privacy+protection+using+precompiled+contracts+for+ring+signature+verification__abbrev.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+## Preamble
+
+    EIP:
```
Comment (axic) [inline EIPS/eip-draft_Transferring+ethers+with+privacy+protection+using+precompiled+contracts+for+ring+signature+verification__abbrev.md]: This file needs to go into a separate PR to have a number allocated.

**Comment by axic** [comment]
@JackXLu are you still pursuing these EIPs?
